BSManagedDocument.m

```diff
@@ -1470,31 +1388,27 @@ - (NSDocument *)duplicateAndReturnError:(NSError **)outError;
      with ARC in macOS 10.15.)  Anyhow, initializing variables is always a
      good practice!  */
     
-    // If the doc is brand new, have to force the autosave to write to disk
-    if (!self.fileURL && !self.autosavedContentsFileURL && !self.hasUnautosavedChanges)
-    {
-        [self updateChangeCount:NSChangeDone];
-        NSDocument *result = [self duplicateAndReturnError:outError];
-        [self updateChangeCount:NSChangeUndone];
-        return result;
-    }
-    
+    // Accessing the MOC ensures a backing store exists
+    (void)[self managedObjectContext];
     
     // Make sure copy on disk is up-to-date
     if (![self fakeSynchronousAutosaveAndReturnError:outError]) return nil;
     
-    
     // Let super handle the overall duplication so it gets the window-handling
-    // right. But use custom writing logic that actually copies the existing doc
-    BOOL (^writingBlock)(NSURL*, NSSaveOperationType, NSURL*, NSError**) = ^(NSURL *url, NSSaveOperationType saveOperation, NSURL *originalContentsURL, NSError **error) {
-        return [self writeBackupToURL:url error:error];
-    };
+    // The writing itself occurs as an "autosave elsewhere" to somewhere other
+    // than the autosavedContentsFileURL
+    return [super duplicateAndReturnError:outError];
+}
+
+- (void)shareDocumentWithSharingService:(NSSharingService *)sharingService completionHandler:(void (^)(BOOL))completionHandler API_AVAILABLE(macos(10.13)) {
+    // Accessing the MOC ensures a backing store exists
+    (void)[self managedObjectContext];
     
-    self.writingBlock = writingBlock;
-    NSDocument *result = [super duplicateAndReturnError:outError];
-    self.writingBlock = nil;
+    // Make sure copy on disk is up-to-date
+    if (![self fakeSynchronousAutosaveAndReturnError:nil]) return;
```

Comment:
I wonder whether this branch ought to call `completionHandler(NO)` to indicate failure to the caller.

(It's strange that the API does not provide a way to return an error.)